### PR TITLE
static-web-server: init at 2.14.1

### PR DESCRIPTION
**Pull request by figsoda**
###### Description of changes

https://sws.joseluisq.net
https://github.com/static-web-server/static-web-server

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/static-web-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  checkFlags = [
+    # TODO: investigate why these tests fail
+    "--skip=tests::handle_byte_ranges_if_range_too_old"
+    "--skip=tests::handle_not_modified"
+    "--skip=handle_precondition"
+  ];
+
+  meta = with lib; {
+    description = "An asynchronus web server for static files-serving.";
```
Comment (sikmir) [inline pkgs/servers/static-web-server/default.nix:28]:
```suggestion
    description = "An asynchronus web server for static files-serving";
```

Comment (figsoda) [reply pkgs/servers/static-web-server/default.nix]: thanks, fixed it

**Comment by clerie** [comment]
Result of `nixpkgs-review pr 206662` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>static-web-server</li>
  </ul>
</details>

**Comment by clerie** [comment]
Huh, merged before finishing review 😅
Anyway, started server and it returned content as expected :)

**Comment by joseluisq** [comment]
SWS maintainer here.
Just a bit late thanks to all you guys and specially @figsoda for the effort to incorporate SWS into nixpkgs ⚡
It should be mentioned in the installation doc section soon.

**Pull request merged**